packages/analytics-js/src/components/pluginsManager/PluginsManager.ts

```diff
@@ -94,101 +96,117 @@ class PluginsManager implements IPluginsManager {
+        configurationStatusStr: 'Error reporting is enabled',
+        activePluginName: state.reporting.errorReportingProviderPluginName.value,
+        basePlugins: ['ErrorReporting'],
+        supportedPlugins: Object.values(ErrorReportingProvidersToPluginNameMap),
+      },
+      {
+        configurationStatus: () =>
+          getNonCloudDestinations(state.nativeDestinations.configuredDestinations.value ?? [])
+            .length > 0,
+        configurationStatusStr: 'Device mode destinations are connected to the source',
+        supportedPlugins: ['DeviceModeDestinations', 'NativeDestinationQueue'] as PluginName[],
+      },
+      {
+        configurationStatus: () =>
+          getNonCloudDestinations(state.nativeDestinations.configuredDestinations.value ?? []).some(
+            destination => destination.shouldApplyDeviceModeTransformation,
           ),
-      );
-    }
-
-    // Cloud mode (dataplane) events delivery plugins
-    if (state.loadOptions.value.useBeacon === true && state.capabilities.isBeaconAvailable.value) {
-      pluginsToLoadFromConfig = pluginsToLoadFromConfig.filter(
-        pluginName => pluginName !== 'XhrQueue',
-      );
-    } else {
-      if (state.loadOptions.value.useBeacon === true) {
-        this.logger?.warn(UNSUPPORTED_BEACON_API_WARNING(PLUGINS_MANAGER));
+        configurationStatusStr:
+          'Device mode transformation is enabled for at least one destination',
+        supportedPlugins: ['DeviceModeTransformation'] as PluginName[],
+      },
+      {
+        configurationStatus: () => isDefined(state.consents.activeConsentManagerPluginName.value),
+        configurationStatusStr: 'Consent management is enabled',
+        activePluginName: state.consents.activeConsentManagerPluginName.value,
+        supportedPlugins: Object.values(ConsentManagersToPluginNameMap),
+      },
+      {
+        configurationStatus: () => isDefined(state.storage.encryptionPluginName.value),
+        configurationStatusStr: 'Storage encryption is enabled',
+        activePluginName: state.storage.encryptionPluginName.value,
+        supportedPlugins: Object.values(StorageEncryptionVersionsToPluginNameMap),
+      },
+      {
+        configurationStatus: () => state.storage.migrate.value,
+        configurationStatusStr: 'Storage migration is enabled',
+        supportedPlugins: ['StorageMigrator'],
+      },
+    ];
+
+    const addMissingPlugins = false;
+    pluginGroupsToProcess.forEach(group => {
+      if (group.configurationStatus()) {
+        pluginsToLoadFromConfig = pluginsToLoadFromConfig.filter(
+          group.activePluginName
+            ? pluginName =>
+                !(
+                  pluginName !== group.activePluginName &&
+                  group.supportedPlugins.includes(pluginName)
+                )
+            : pluginName => isDefined(pluginName), // pass through
+        );
+
+        this.addMissingPlugins(group, addMissingPlugins, pluginsToLoadFromConfig);
```

Comment:
Why do we need to call the addMissingPlugins fn for each plugin group?
add a checking if shouldAddMissingPlugins is true then only call this fn.

Reply:
This function takes care of warning the users now but can also add the missing plugins if needed.